api/README.md

```diff
@@ -1 +1,269 @@
+
+### Actions
+
+After the resources are defined you need to find out what kind of actions should be applied to them. The basic CRUD actions can be omitted by simply using the HTTP request methods mentioned above. If an action doesn't fit into the CRUD operations there are two main approaches you can choose from:
+
+1. Treat it like a sub-resource and modify it with CRUD operations. E.g. `PUT /posts/31/like` to like the post and `DELETE /posts/31/like` to dislike the post.
+1. There are cases when you can't map the action to a single resource. In this case naming the endpoint with the name of the action makes the most sense, e.g. `/search`, `/sign-in`, etc.
+
+**You should never ever** leak any implementation details via your API.
+
+## HTTP Request Methods
+
+The HTTP request methods `GET`, `POST`, `PUT`, `PATCH` and `DELETE` have specific meaning in manipulating these.
+
+### GET
+
+- Retrieves a resource or a list of resources.
+- It should **never** update a resource.
+- Shouldn't even update the access logs.
+- Use query strings instead of request body.
+
+To retrieve the list of posts filtered by the keyword "awesome":
+```HTTP
+GET /posts?search=awesome
+```
+
+To retrieve the post with id 31:
+```HTTP
+GET /posts/31
+```
+
+### POST
+
+- Creates a resource.
+- It should be used when the server has to generate a unique key for the created resource.
+- Multiple POST requests to the same URL create duplicate records.
+
+To create a post:
+```HTTP
+POST /posts
+Request payload:
+{
+  "title": "Awesome title",
+  "content": "even more awesomeness"
+}
+```
+
+> The newly created resource (with a unique identifier) should be returned in the response body.
+
+### PUT
+
+- Creates/updates a resource.
+- Multiple PUT requests to the same URL don't create/update duplicate records.
+- It should be used when the client is aware of the url of the resource.
+
+To update the post with id 31:
+```HTTP
+PUT /posts/31
+Request payload:
+{
+  "title": "Awesome title",
+  "content": "even more awesomeness"
+}
+```
+
+> If creating/updating the resource doesn't trigger unpredicted changes in the underlaying data there is no need to return the resource in the response body.
+
+### PATCH
+
+- Partially updates a resource.
+- Updated attributes of the resource should be provided only.
+
+To update the content of the post with id 31:
+```HTTP
+PATCH /posts/31
+Request payload:
+{
+  "content": "less awesomeness"
+}
+```
+
+> No response body should be returned.
+
+### DELETE
+
+- Deletes a resource.
+
+To delete the post with id 31:
+```HTTP
+DELETE /posts/31
+```
+
+> No response body should be returned.
+
+## URL Naming
+
+It is a good practice to have all resources and actions written in `lowercase-separated-by-hyphens`, e.g. `/sign-in`, `/reset-password`, etc.
+
+## Response
```

Comment:
Might not be directly related to the API standards but we should somehow aim to use the same structure and format when building JSON objects for the responses. I have seen quite some responses throughout the years, especially with beginners.